mono/metadata/boehm-gc.c

```diff
@@ -529,21 +529,21 @@ register_root (gpointer arg)
 }
 
 int
-mono_gc_register_root (char *start, size_t size, void *descr, MonoGCRootSource source, const char *msg)
+mono_gc_register_root (char *start, size_t size, void *descr, MonoGCRootSource source, void *key, const char *msg)
 {
 	RootData root_data;
 	root_data.start = start;
 	/* Boehm root processing requires one byte past end of region to be scanned */
 	root_data.end = start + size + 1;
 	GC_call_with_alloc_lock (register_root, &root_data);
-
+	MONO_PROFILER_RAISE (root_register, (start, size, source, key, msg));
```

Comment:
I don't think we should raise these events in Boehm since we don't report roots anyway.